object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -146,7 +146,12 @@ void checkMustCallInvoked(ControlFlowGraph cfg) {
 
   private void handleTernary(Node node, Set<ImmutableSet<LocalVarWithTree>> defs) {
     LocalVariableNode ternaryLocal = typeFactory.getTempVarForTree(node);
-
+    // This can happen when one side of the ternary is null, I think.
+    // Without this, we get a crash in tests/mustcall/ZookeeperTernaryCrash.java
+    // on the only ternary in that program snippet.
+    if (ternaryLocal == null) {
+      return;
```

Comment:
I think we are creating a temporary variable for all ternary expressions in ObjectConstructionTransfer.java, so why is it null in this case?

Reply:
I don't know. I suspect it might have something to do with wildcards?

Reply:
or possibly that the other side of that ternary is `null`?

Reply:
But we could handle [this](https://github.com/kelloggm/object-construction-checker/blob/e51bc0304283c5734be161df7b096c06df66c9b6/object-construction-checker/tests/mustcall/ACRegularExitPointTest.java#L349) test! In this test case null in then operand.

Reply:
We don't add a temporary varibale for primitive types into the store ([here](https://github.com/kelloggm/object-construction-checker/blob/orElseNull/object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionTransfer.java#L159)) and that's why sometimes `ternaryLocal` is null. In this example:
`final Integer type = entry.size() >= 2 ? (Integer) entry.get(0) : null;`
ternary expression has primitive type in the then branch and we don't have a temporary vairbale for it in the store.